app/javascript/lib/validation/validate-file-content.js

```diff
@@ -348,15 +403,25 @@ async function parseFile(file, fileType) {
       if (lines.length < 2) {
         return { table, delimiter, issues: [['error', 'cap:format:no-newlines', 'File does not contain newlines to separate rows']] }
       }
+    }
+    if (['Cluster', 'Metadata', 'Expression Matrix'].includes(fileType)) {
       // if there are no encoding issues, and this isn't a gzipped file, validate content
       delimiter = sniffDelimiter(headerLines, mimeType)
       table = []
       for (let i = 0; i < lines.length; i++) {
         table.push(lines[i].split(delimiter))
       }
-      const headerTable = table.slice(0, 2)
-      issues = await validateCapFormat(headerTable, fileType)
-      issues = issues.concat(validateUniqueCellNamesWithinFile(table))
+
+      if (fileType === 'Expression Matrix') {
+        const firstRowTable = table.slice(0, 1)
+        issues = issues.concat(validateUniqueCellNamesWithinFile(firstRowTable[0], fileType))
+        issues = issues.concat(validateGeneInHeader(firstRowTable))
+        issues = issues.concat(validateColumnNumberAndValues(table))
+      } else if (['Cluster', 'Metadata'].includes(fileType)) {
+        const headerTable = table.slice(0, 2)
+        issues = await validateCapFormat(headerTable, fileType)
+        issues = issues.concat(validateUniqueCellNamesWithinFile(table, fileType))
+      }
```

Comment:
I'd organize this differently, for greater simplicity, coverage, and sync CSFV extensibility.

* `validateUniqueCellNamesWithinFile` applies to all file types handled here.  Per your changes in this PR, that function now has special handling for dense matrices, so this special-casing for the first argument (`firstRowTable` / `table`) seems more coherent to do within that function.

* `validateGeneInHeader` is fundamentally a cap validation, i.e. it requires only the first line(s) in a file.  Thus, it'll be a good candidate for sync CSFV.  Grouping all cap validations into `validateCapFormat` would let us more easily apply all of them to sync.  So like we can handle file type-specific validations _within_ `validateUniqueCellNamesWithinFile`, I'd do similarly in `validateCapFormat`.  (I'd also update its docstring to note that a cap is the first "few" lines, and applies to all the main SCP file types.)

* `validateColumnNumberAndValues` should apply to all three file types handled here.

Apologies that I can't be more specific; the GitHub review UI is limiting.  I'm happy to chat if it'd help!  The above notes would simplify this block to:

```suggestion
      const cap = table.slice(0, fileType === 'Expression Matrix' ? 1 : 2)
      
      issues = await validateCapFormat(cap, fileType)
      issues = issues.concat(validateUniqueCellNamesWithinFile(table, fileType))
      issues = issues.concat(validateColumnNumberAndValues(table))
```